### PR TITLE
Adding support for light vs dark logos for logo cobranding

### PR DESCRIPTION
**Pull request by JakeBat**
# Problem

dark cobranding logos weren't displaying on screens with dark backgorounds

# Solution

add options for light and dark logos for cobranding, similar to the mobile sdks

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

File: licenses.json [hunk 1]
```diff
@@ -9027,7 +9036,7 @@
     "publisher": "Web SDK Customer Support",
     "email": "web-sdk@onfido.com",
     "url": "https://github.com/onfido",
-    "path": "onfido-sdk-ui",
+    "path": "/Users/jake.batchelor/WebstormProjects/onfido-sdk-ui",
```
Comment (kopijunkie) [inline licenses.json:9039]:
```suggestion
    "path": "onfido-sdk-ui",
```

**Review by kopijunkie — APPROVED** [review]
Looks fine code wise to me aside from the minor path edit suggestion in `licenses.json`. We can fix this before release anyway if needed, pending acceptance. 
BTW is there a ticket to track this fix on your end?

File: src/components/App/ModalApp.tsx [hunk 2]
```diff
@@ -293,6 +293,14 @@ class ModalApp extends Component<Props> {
   ) => {
     if (isValidEnterpriseFeature) {
       this.props.actions.showLogoCobranding(logoCobrandConfig)
+      document.documentElement.style.setProperty(
+        '--darkLogoSrc',
```
Comment (stefaniacardenas) [inline src/components/App/ModalApp.tsx:297]: We are adding a property to the host app `html`, which I don't think is ideal. Also, the property name is not generic enough IMO. Can we add this property just to the onfido wrapper?

**Review by stefaniacardenas — CHANGES_REQUESTED** [review]

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (kopijunkie) [reply src/components/App/ModalApp.tsx:297]: Agree with @stefaniacardenas comments above. 
An alternative is to do something similar to how customisation options are applied. See `setUICustomizations` in `src/components/Theme/utils.ts`.

**Review by kopijunkie — CHANGES_REQUESTED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 3]
```diff
@@ -228,6 +228,14 @@ export default class CrossDeviceMobileRouter extends Component<
...
         ) {
           this.props.actions.showLogoCobranding(enterpriseFeatures.logoCobrand)
+          document.documentElement.style.setProperty(
+            '--darkLogoSrc',
+            `url(${enterpriseFeatures.logoCobrand.darkLogoSrc})`
+          )
+          document.documentElement.style.setProperty(
+            '--lightLogoSrc',
+            `url(${enterpriseFeatures.logoCobrand.lightLogoSrc})`
+          )
```
Comment (stefaniacardenas) [inline src/components/Router/CrossDeviceMobileRouter.tsx:238]: You can reuse the `setCobrandingLogos` function here

Comment (JakeBat) [reply src/components/Router/CrossDeviceMobileRouter.tsx]: Updated

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Review by stefaniacardenas — APPROVED** [review]
LGTM. We can create a ticket to refactor this so that we can easily pass JS variables to scss files

File: src/components/App/ModalApp.tsx (hunk 2 above)
Comment (JakeBat) [reply src/components/App/ModalApp.tsx]: Updated

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/utils.ts [hunk 4]
```diff
@@ -12,6 +13,14 @@ export const setUICustomizations = (customUI: UICustomizationOptions): void => {
   }
 }
 
+export const setCobrandingLogos = (
+  logoCobrandConfig: EnterpriseLogoCobranding
+): void => {
+  const style = document.createElement('style')
+  style.textContent = `:root { --darkLogoSrc: url(${logoCobrandConfig.darkLogoSrc}); \n --lightLogoSrc: url(${logoCobrandConfig.lightLogoSrc});}`
```
Comment (kopijunkie) [inline src/components/Theme/utils.ts:20]: I think the CSS variable name needs to be less generic, something like `--osdk-dark-cobrand-logo-src` and `--osdk-light-cobrand-logo-src` (kebabcase for consistency with the UI customisation CSS variables' convention). On the off chance the client has a `--darkLogoSrc`, `--lightLogoSrc`.

**Review by kopijunkie — CHANGES_REQUESTED** [review]

File: src/components/Theme/utils.ts (hunk 4 above)
Comment (JakeBat) [reply src/components/Theme/utils.ts]: Updated

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

**Review by kopijunkie — APPROVED** [review]
LGTM

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1389-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1389-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**